johnsonandjohnson/Bodiless-JS
# Bug/fix resizing 598

## Changes


* Depricated FlowContainer prop defaultWidth
* Created FlowContainer prop defaultWidthClasses
* Ensure that class are added to the dom as things are resized
* Ensure that the width is removed when resizing ends

## Test Instructions

Verify in Edit/Review/Static
1. Verify anything added by default flowcontainer automatically gets 'w-full' 
1. Verify resizing anything in the flowcontainer maintains w-full and size set (Desktop -- all sizes, Tablet - 50%, Mobile - none) 
1. Place a varity of container items 
* Row with 25% at desktop, 50% at tablet, 100% at mobile
* Row with 50% at tablet & desktop, 100 at mobile 
* Row with 100% at mobile, 50% at tablet, 100% at desktop

Verify Edit mode only:
1. Verify that constrained widths of 50/100 are still in effect. (Desktop, Tablet, Mobile)
1. Verify that limit is in effect
1. Verify that each of the default widths on placement holds true

Verify in Edit/Review/Static
1. Verify https://github.com/johnsonandjohnson/Bodiless-JS/issues/327
* Add 4 accordions and add titles
* leave first two 1005
* Resize to tablet and make 2 50% at tablet and thus desktop.
* All should render at 100% on mobile

Verify in Edit/Review/Static
* Add a new page.
* Add 3 image components to page
* Resize to tablet & resize each to 33%.
* Click out of edit & resize to desktop
* Click edit and they should be at 33%

## Related Issues

Fixes #598 

BREAKING CHANGE: 
This PR breaks the old FlowContainer API's prop defaultWidth, it uses to take a number and now needs to take a function that will return a string.
